### PR TITLE
Add the ability to upload a build from a local file path

### PR DESCRIPTION
**Pull request by Blooym**
Ticket: https://app.clickup.com/t/86784y8wc

Depends on: moropo-com/platform#1239

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
     required: false
   github_token:
-    description: "GitHub access token (optional) - so that the action can comment on the PR with the test trigger feedback"
+    description: 'GitHub access token (optional) - so that the action can comment on the PR with the test trigger feedback'
     required: false
     default: ${{ github.token }}
+  build_path:
+    description: 'Path to the build artifact to upload to Moropo before running the test'
+    required: false
+  upload_build:
```
Comment (Foosballfan) [inline action.yml]: Is this meaningful? passing in a build path should be enough to infer that we want to upload it

File: index.ts [hunk 2]
```diff
@@ -44,52 +53,96 @@ const buildMessageString = ({
...
+    const scheduledTestRunId = getInput('scheduled_test_id');
+    const apiKey = getInput('api_key');
+    const githubToken = getInput('github_token');
+    const buildPath = getInput('build_path');
+    const moropoUrl = new URL(getInput('moropo_url'));
+    const moropoApiUrl = new URL(getInput('moropo_api_url'));
+
+    if (!moropoUrl || !moropoApiUrl || !apiKey || !scheduledTestRunId) {
+      setFailed(
+        'Missing one or more required inputs: moropo_url, moropo_api_url, api_key, scheduled_test_id'
```
Comment (Foosballfan) [inline index.ts]: probably shouldnt include moropo urls in here

Comment (Blooym) [reply index.ts]: They're automatically considered present as they are set as defaults in the `action.yml`

**Review by Foosballfan — APPROVED** [review]
Please update readme with build upload info

File: index.ts (hunk 2 above)
Comment (Foosballfan) [reply index.ts]: it would confuse a user if they dont provide apikey or test run id but then get told that moropo url is missing

**Pull request merged**